lib/iris/coords.py

```diff
@@ -497,6 +513,14 @@ def is_full_slice(s):
                 if bounds is not None:
                     bounds = bounds[keys + (Ellipsis, )]
 
+        # Copy data after indexing, if needed, to avoid making coords that are
+        # views on other coords.  This will not realise lazy data.
+        if not points_copied:
+            points = points.copy()
```

Comment:
@pp-mo doesn't this mean we end up effectively copying twice when combined with [L552](https://github.com/SciTools/iris/pull/2517/files#diff-fae2e3bb7827f6df8671256d1c779951R552)?